src/BuildingBlocks/EventBus/EventBusRabbitMQ/DefaultRabbitMQPersistentConnection.cs

```diff
@@ -1,131 +1,120 @@
+﻿namespace Microsoft.eShopOnContainers.BuildingBlocks.EventBusRabbitMQ;
+
+public class DefaultRabbitMQPersistentConnection
+    : IRabbitMQPersistentConnection
 {
-    public class DefaultRabbitMQPersistentConnection
-       : IRabbitMQPersistentConnection
-    {
-        private readonly IConnectionFactory _connectionFactory;
-        private readonly ILogger<DefaultRabbitMQPersistentConnection> _logger;
-        private readonly int _retryCount;
-        IConnection _connection;
-        bool _disposed;
+    private readonly IConnectionFactory _connectionFactory;
+    private readonly ILogger<DefaultRabbitMQPersistentConnection> _logger;
+    private readonly int _retryCount;
+    IConnection _connection;
+    bool _disposed;
 
-        object sync_root = new object();
+    object sync_root = new object();
 
-        public DefaultRabbitMQPersistentConnection(IConnectionFactory connectionFactory, ILogger<DefaultRabbitMQPersistentConnection> logger, int retryCount = 5)
-        {
-            _connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));
-            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            _retryCount = retryCount;
-        }
+    public DefaultRabbitMQPersistentConnection(IConnectionFactory connectionFactory, ILogger<DefaultRabbitMQPersistentConnection> logger, int retryCount = 5)
+    {
+        _connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _retryCount = retryCount;
+    }
 
-        public bool IsConnected
+    public bool IsConnected
+    {
+        get
         {
-            get
-            {
-                return _connection != null && _connection.IsOpen && !_disposed;
-            }
+            return _connection != null && _connection.IsOpen && !_disposed;
         }
+    }
 
-        public IModel CreateModel()
+    public IModel CreateModel()
+    {
+        if (!IsConnected)
         {
-            if (!IsConnected)
-            {
-                throw new InvalidOperationException("No RabbitMQ connections are available to perform this action");
-            }
-
-            return _connection.CreateModel();
+            throw new InvalidOperationException("No RabbitMQ connections are available to perform this action");
         }
 
-        public void Dispose()
-        {
-            if (_disposed) return;
+        return _connection.CreateModel();
+    }
 
-            _disposed = true;
+    public void Dispose()
+    {
+        if (_disposed) return;
 
-            try
-            {
-                _connection.Dispose();
-            }
-            catch (IOException ex)
-            {
-                _logger.LogCritical(ex.ToString());
-            }
+        _disposed = true;
+
+        try
+        {
+            _connection.Dispose();
+        }
```

Comment:
```suggestion
        {
            _connection.ConnectionShutdown -= OnConnectionShutdown;
            _connection.CallbackException -= OnCallbackException;
            _connection.ConnectionBlocked -= OnConnectionBlocked;
            _connection.Dispose();
        }
```

Unwire event handlers when performing clean up.